### PR TITLE
chore: add troubleshooting docs, cleanup related docs

### PR DESCRIPTION
**Pull request by Gankra**

### CODE REVIEW

File: book/src/troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+2. update and repair: [run `cargo dist init` again](./updating.md)
+3. test your process: [try `pr-run-mode = "upload"`](./ci/github.md#build-and-upload-artifacts-on-every-pull-request)
+
+These are also great steps to follow proactively, if you're updating your cargo-dist config.
+
+
+
+## Nothing To Release / Missing Packages / Too Many Packages
+
+cargo-dist tries to support as many release workflows as possible, and that means it needs you to tell it what you're interested in releasing. There are several ways to opt things in and out of being released, the most important are:
```
Comment (mistydemeo) [inline book/src/troubleshooting.md:22]: Comma splice

```suggestion
cargo-dist tries to support as many release workflows as possible, and that means it needs you to tell it what you're interested in releasing. There are several ways to opt things in and out of being released; the most important are:
```

File: book/src/troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+We *DO NOT* recommend "retry all jobs". Either it's redundant or it can cause problems with trying to repeat side-effects like publishing a package.
+
+
+### Failed CI: Busted Builds
+
+If you believe something was busted in your release process, and the commit you tried to release from isn't suitable, that's generally ok! Usually this will occur because some part of your build is broken, perhaps only when releasing with cargo-dist, or only on a particular platform.
+
+If this is the case then presumably your release process errored out before the the "host" step where we actually uploaded anything, so good news: no side effects need to be rolled back!
+
+Well, one side-effect needs to be dealt with, but it's the one you did to kick off the release: [delete the git tag from github and your local machine](https://stackoverflow.com/questions/5480258/how-can-i-delete-a-remote-tag), get your build sorted out, and then tag the new release commit.
```
Comment (mistydemeo) [inline book/src/troubleshooting.md:54]: We should avoid telling people to delete and retag, IMO.

File: cargo-dist/src/errors.rs [hunk 3]
```diff
@@ -171,7 +171,7 @@ pub enum DistError {
     },
     /// Not an error; indicates that a file's contents differ via --check
     #[error("{} has out of date contents and needs to be regenerated:\n{diff}", file.origin_path())]
-    #[diagnostic(help("run 'cargo dist init' to update the file or set 'allow-dirty' in Cargo.toml to ignore out of date contents"))]
+    #[diagnostic(help("run 'cargo dist init' to update the file\n( 'allow-dirty' in Cargo.toml to ignore out of date contents)"))]
```
Comment (mistydemeo) [inline cargo-dist/src/errors.rs:174]: No space after opening paren, IMO

```suggestion
    #[diagnostic(help("run 'cargo dist init' to update the file\n('allow-dirty' in Cargo.toml to ignore out of date contents)"))]
```

File: book/src/reference/artifact-url.md [hunk 4]
```diff
@@ -1,16 +1,21 @@
 # Artifact URL
 
+<!-- toc -->
+
 [Fetching installers](../installers/index.md) need to know where to download the [actual binaries](../artifacts/archives.md) from, so cargo-dist needs to be aware of a base Artifact URL that it can derive download URLs from when it builds those kinds of installers.
 
-## GitHub
+Currently artifact URLs are derived from [hosting providers](./config#hosting) and [source hosts](#source-hosts). Both must be well-defined for fetching installers to be enabled. Here is the behaviour of each hosting provider in more excruciating detail:
```
[the remark this answers is not in the document]
Comment (Gankra) [reply book/src/reference/artifact-url.md:7]: Is the first paragraph not that? (anything in mind?)

File: book/src/troubleshooting.md (hunk 2 above)
Comment (Gankra) [reply book/src/troubleshooting.md:54]: I don't love it either but people get very precious about version numbers, so if we don't explain it they might mess things up worse.

Comment (mistydemeo) [reply book/src/troubleshooting.md:54]: Yeah, that's fair.

**Review by ashleygwilliams — APPROVED** [review]

**Pull request merged**